Themes/default/scripts/suggest.js

```diff
@@ -338,24 +282,40 @@ smc_AutoSuggest.prototype.removeLastSearchString = function ()
 smc_AutoSuggest.prototype.addItemLink = function (sItemId, sItemName, bFromSubmit)
 {
 	// Increase the internal item count.
-	this.iItemCount ++;
+	this.iItemCount++;
 
 	// If there's a callback then call it.
-	if ('oCallback' in this && 'onBeforeAddItem' in this.oCallback && typeof(this.oCallback.onBeforeAddItem) == 'string')
+	if (typeof this.oCallback.onBeforeAddItem == 'function')
 	{
 		// If it returns false the item must not be added.
-		if (!eval(this.oCallback.onBeforeAddItem + '(' + this.opt.sSelf + ', \'' + sItemId + '\');'))
+		if (!this.oCallback.onBeforeAddItem.call(this, sItemId))
+			return;
+	}
+	// Backward compatibility; to be removed in the future
```

Comment:
This line is too vague. It needs to specify a milestone.

```suggestion
	// Backward compatibility; to be removed in v4.0.0.
```

Opinions? I hereby suggest the next major version.